gateway/starknet_test.go

```diff
@@ -9,23 +9,26 @@ import (
 	"os"
 	"os/exec"
 	"testing"
+
+	"github.com/dontpanicdao/caigo"
+	"github.com/dontpanicdao/caigo/types"
 )
 
 func TestExecuteGoerli(t *testing.T) {
-	curve, err := SC(WithConstants("./pedersen_params.json"))
+	curve, err := caigo.SC(caigo.WithConstants("./pedersen_params.json"))
 	if err != nil {
 		t.Errorf("Could not init with constant points: %v\n", err)
 	}
 
 	priv, _ := new(big.Int).SetString("3904bd288b88a1dcd73e648b10642d63cb9b2ffd86526deee9d073f0690139e", 16)
 	x, y, _ := curve.PrivateToPoint(priv)
 
-	signer, err := curve.NewSigner(priv, x, y, NewGateway())
+	signer, err := curve.NewSigner(priv, x, y, NewProvider())
```

Comment:
`./starknet_test.go:26:45: cannot use NewProvider() (value of type *GatewayProvider) as type types.Provider in argument to curve.NewSigner:
        *GatewayProvider does not implement types.Provider (wrong type for BlockByHash method)
                have BlockByHash(ctx context.Context, hash string) (*types.Block, error)
                want BlockByHash(context.Context, string, string) (*types.Block, error)`